### ISSUE TITLE
`ca-es-valencia` locale data is duplicated on intl-numberformat

### ISSUE THREAD

**Issue by syAndyDev** [formatjs/formatjs#3377]
## Which package?
`intl-numberformat@7.4.0`
## Describe the bug
Duplicated files generated inside the package. cause of this, these warnings are produced.

**!!NOTE**: I replaced my project folder with `###`
```bash
./node_modules/@formatjs/intl-numberformat/locale-data/ca-ES-VALENCIA.d.ts
There are multiple modules with names that only differ in casing.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Use equal casing. Compare these module identifiers:
* ###/node_modules/@formatjs/intl-numberformat/locale-data/ca-ES-VALENCIA.d.ts
    Used by 2 module(s), i. e.
   ###/node_modules/@formatjs/intl-numberformat/locale-data lazy /^\.\/.*$/ groupOptions: {} namespace object
* ###/node_modules/@formatjs/intl-numberformat/locale-data/ca-ES-valencia.d.ts
    Used by 2 module(s), i. e.
    ###/node_modules/@formatjs/intl-numberformat/locale-data lazy /^\.\/.*$/ groupOptions: {} namespace object
./node_modules/@formatjs/intl-numberformat/locale-data/ca-ES-VALENCIA.js
There are multiple modules with names that only differ in casing.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Use equal casing. Compare these module identifiers:
* ###/node_modules/react-scripts/node_modules/babel-loader/lib/index.js??ref--5-oneOf-3!###/node_modules/@formatjs/intl-numberformat/locale-data/ca-ES-VALENCIA.js
    Used by 2 module(s), i. e.
   ###/node_modules/@formatjs/intl-numberformat/locale-data lazy /^\.\/.*$/ groupOptions: {} namespace object
* ###/node_modules/react-scripts/node_modules/babel-loader/lib/index.js??ref--5-oneOf-3!###/node_modules/@formatjs/intl-numberformat/locale-data/ca-ES-valencia.js
    Used by 2 module(s), i. e.
    ###/node_modules/@formatjs/intl-numberformat/locale-data lazy /^\.\/.*$/ groupOptions: {} namespace object
```

### Reproducible Steps/Repo

1. install latest formatjs modules
```
    "@formatjs/intl-getcanonicallocales": "^1.9.0",
    "@formatjs/intl-locale": "^2.4.43",
    "@formatjs/intl-numberformat": "^7.4.0",
    "@formatjs/intl-pluralrules": "^4.3.0",
```
2. then build or run with dev mode.

## Expected behavior
Duplicated files should be removed.

## Screenshots
![image](https://user-images.githubusercontent.com/51244659/150085395-6e43147c-6175-443e-92e8-7fe16ccf9205.png)

## Desktop (please complete the following information):

- OS: Ubuntu 21.04
- Browser: chrome
- Version: 97.0.4692.71 (Official Build) (64-bit)

**Comment by longlho** [comment]
hmm looks like there're only 1 version of that (https://unpkg.com/browse/@formatjs/intl-numberformat@7.4.0/locale-data/). I checked by manually download as well.

**Comment by cvkline** [comment]
For what it's worth, this doesn't appear to break any behavior but it is filling up our build logs with warnings complaining about the similar file names differing only in case. It'd be nice to get fixed. This is version 7.4.1 of intl-numberformat, but I'm also seeing it in intl-datetimeformat and intl-relativetimeformat.

![Screen Shot 2022-02-04 at 12 09 35 PM](https://user-images.githubusercontent.com/6576592/152580864-74c4b4c3-f5ec-4a7c-a85b-927e723f2bcb.png)

**Comment by cvkline** [comment]
AHA, I found the issue, and I am guessing it is what everyone else is encountering as well. I wound up with the two filenames `ca-ES-valencia.js` and `ca-ES-VALENCIA.js` after up-revving @formatjs in my package.json and rerunning `npm` _but_ I did not clear out my project's `node_modules` directory first. I think in previous versions it was capitalized and now it's in lowercase, and npm just isn't removing the old files.

The fix is to `rm -rf node_modules` and re-running npm to get a fresh install.

Recommend this be closed given that fix.
